File: dev-docs/snippets/documents/get.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Fetch a document's metadata from the server.
export async function get(client: JsBaoClient, documentId: string) {
  // #region example
  const info = await client.documents.get(documentId);
  // #endregion example
  return info; // info.title, info.lastModified, info.permission, …
}

File: dev-docs/snippets/documents/get.swift
import JsBaoClient

// Fetch a document's metadata from the server as a typed `DocumentInfo`.
func get(client: JsBaoClient, documentId: String) async throws {
  // #region example
  let info = try await client.documents.get(documentId: documentId)
  let title = info.title
  let lastModified = info.lastModified
  // #endregion example
  _ = (title, lastModified)
}
